app/src/main/java/com/github/jorgecastillo/kotlinandroid/functional/Future.kt

```diff
@@ -1,35 +1,68 @@
 package com.github.jorgecastillo.kotlinandroid.functional
 
+import kategory.*
 import kotlinx.coroutines.experimental.CommonPool
 import kotlinx.coroutines.experimental.Deferred
 import kotlinx.coroutines.experimental.android.UI
 import kotlinx.coroutines.experimental.async
 import kotlinx.coroutines.experimental.launch
 
+typealias FutureKind<A> = HK<Future.F, A>
+
+fun <A> FutureKind<A>.ev(): Future<A> =
+        this as Future<A>
+
 /**
  * Basic future implementation to achieve asynchronicity based on Kotlin coroutines.
  */
-class Future<T> {
+class Future<T> : FutureKind<T> {
```

Comment:
I would try to not introduce Bandhook Result here if it's not solving any other issues we have. If that's not the case, it's ok. But I would like to keep the samples as neat and straightforward as possible to showcase which problems are being resolved and not any single line of extra code that would probably be used in some future (or probably not), since that would just add noise, IMO.

This branch is expected to be a didactic case of tagless final usage over Android. So let's wait for `kategory#IO` so we can get rid of the future here and simplify the code a client needs to write. Also, should we move `AsyncResult` to a different module inside kategory then?